core/store/src/trie/iterator.rs

```diff
@@ -424,19 +426,42 @@ impl<'a> Iterator for TrieIterator<'a> {
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use std::collections::BTreeMap;
+pub enum TrieIterator<'a> {
+    Disk(DiskTrieIterator<'a>),
+    Memtrie(MemTrieIterator<'a>),
```

Comment:
I suspect it is going to be (appreciably) cheaper to use dynamic dispatch over an enum here. Right now iteration needs to match on the enum on each iteration (and since its a `&mut self`, the compiler will most likely have to assume the enum changes for the purposes of various analyses, unless inlining occurs.)

Reply:
Possibly, but this may be premature optimization because the operations being performed while iterating (disk lookups, memtrie decoding, and whatever work the callers is doing with the elements) is much heavier than any sort of dispatch.